text/0419-cdk-launchpads.md

```diff
@@ -0,0 +1,629 @@
+    However, we are not 100% certain the features provided here will be enough to unblock customers who cannot use the CDK today.
+    That's why the implementation plan (see below) contains a Beta period.
+
+    Because of its size, scope and ambiguity, the project also has a high degree of risk.
+
+2. For the restricted security aspect of this RFC, the CDK L2 and higher constructs often create IAM Roles.
+  Because many compliance policies don't allow this, the mechanisms to enforce this on construct authors are fairly ambiguous,
+  not clear how to implement, and pose another risk.
+3. Integrating with AWS Organizations, and having a presence in the AWS Console,
+  will require owning server-side components.
+  Maintaining them will put additional strain on the CDK team.
+
+### What is the technical solution (design) of this feature?
+
+This is a good question, but outside the scope of this document, considering how big the proposed functionality is.
+
+### Is this a breaking change?
+
+No.
+
+### What alternative solutions did you consider?
+
+The alternative is to not create a big project like Launchpads,
+but tackle each aspect (compliance, bootstrapping, shared resources, etc.)
+separately, as its own project.
+
+### What are the drawbacks of this solution?
+
+We will be introducing both a lot of complexity to the already complex CDK CLI,
+and also a separate CLI that we will have to maintain.
+
+It might also require the CDK team to own new server-side components,
+like a new AWS Console, and a new service that integrates with AWS Organizations.
+
+### What is the high-level project plan?
+
+Since this is quite a big and ambiguous project,
+the project plan is a little more detailed than usual.
+
+The plan is divided into the following milestones:
+
+#### 1. Alpha version of automated bootstrapping
+
+In the first version of the Launchpads functionality,
+we want to automate the setup of new accounts in an AWS Organization,
+eliminating the need for running `cdk bootstrap` manually,
+while validating that the approach works before we invest more into it --
+so, we will not have any server-side components the CDK team owns at this stage of the project.
+
+To achieve this goal, we will allow administrators to define a StackSet, using CDK constructs
+(similarly to the [`ProductStack` class](https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-servicecatalog.ProductStack.html)
+in the ServiceCatalog Construct Library).
+We will also create construct(s) that represent the CDK bootstrap resources,
+and allow customizing them.
+
+In order to automatically bootstrap new accounts that get added to a given Organization,
+we will use the [CDK Pipelines library](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.pipelines-readme.html).
+The StackSet will live in a separate Stack, deployed to the master account.
+Any time a new account is added, a CloudWatch Event will trigger the pipeline,
```

Comment:
Unfortunately, the problem is that we don't know what region we need to deploy the StackSet instance to, using only this integration 😕. StackSet are not really very flexible in that regard.

Reply:
When you create the stackset you specify the OU _and_ the regions you want to deploy to right?

Reply:
Hmm, the page you linked to (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stacksets-orgs-manage-auto-deployment.html) doesn't mention choosing a region when enabling the automated deployments?

Reply:
I think it's because it's a feature of _stack sets_ and not specifically automated deployments. You create a stack set (and specify OU/regions) and _then_ you enable/disable automated deployments.

Reply:
I guess that could work, the major downside being that customers would have to decide up front which regions they will bootstrap in (and this decision has to be made both for developer accounts, and for shared environment accounts - they have to be in the same regions).

Reply:
Stack Sets deploy to the same region(s) in all accounts. Not typically something we recommend for users, we recommend they use a different account for every region, so that makes it hard(er).

![image](https://user-images.githubusercontent.com/524162/173711926-5c4d3ec1-8e62-4780-9986-6baa2f818bc2.png)

I'm in favor of using stacksets, but we'll need same additional work to allocate regions and accounts to stacksets.